src/codeu/chat/server/Server.java

```diff
@@ -102,6 +106,10 @@ public void onMessage(InputStream in, OutputStream out) throws IOException {
 
         Serializers.INTEGER.write(out, NetworkCode.NEW_USER_RESPONSE);
         Serializers.nullable(User.SERIALIZER).write(out, user);
+      if (type == NetworkCode.SERVER_INFO_REQUEST) {
```

Comment:
See my above comment - you only need to make that change once, so you can probably delete this section.